mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRV.cpp

```diff
@@ -458,6 +468,39 @@ LogicalResult GPUShuffleConversion::matchAndRewrite(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Rotate
+//===----------------------------------------------------------------------===//
+
+LogicalResult GPURotateConversion::matchAndRewrite(
+    gpu::RotateOp rotateOp, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  auto targetEnv = getTypeConverter<SPIRVTypeConverter>()->getTargetEnv();
```

Comment:
Don't use `auto` here, the type is not obvious based on the context

Reply:
Updated.